packages/@tailwindcss-upgrade/src/codemods/template/is-safe-migration.ts

```diff
@@ -62,10 +70,10 @@ export function isSafeMigration(
   // So let's only skip if we couldn't parse and we are not in Tailwind CSS v3.
   //
   if (!candidate && version.isGreaterThan(3)) {
-    return true
+    return false
```

Comment:
I'm a bit of a dumb dumb here. When we know that a candidate doesn't parse, and we know that we are in v4 and up, then we know that we can't migrate the candidate anyway so we can bail.

That makes it an unsafe migration (which skips all the migrations for this candidate). We were returning `true` which means that we were migrating the candidate. This still worked because eventually the migrations were no-ops because they all bailed.

But let's not do that work in the first place, yeah?

Reply:
This change alone makes the upgrade tool on Tailwind UI's product folder go from ~12s to ~9s

Reply:
Oh and that's because this branch was hit about 1 041 458 times in that codebase.